packages/lib/slots.test.ts

```diff
@@ -210,7 +210,9 @@ describe("Tests the slot logic", () => {
       inviteeDate: dayjs().add(1, "day"),
       frequency: 20,
       minimumBookingNotice: 0,
-      dateRanges: [{ start: dayjs("2021-06-21T00:00:00.000Z"), end: dayjs("2021-06-21T23:45:00.000Z") }],
```

Comment:
In different tz like IST, this will be converted into `2021-06-21T05:30:00+05:30`
We will try to find the first time slot keeping in mind that the availability interval is `20`.
The value of minutes is `30` which ends up eating a slot and failing the test in IST.

Reply:
That is a great diagnosis, but don't we want to keep the test the same and support this feature in IST? This way seems more like we explicitly don't work in IST in order to keep the tests passing - but dateRanges isn't necessarily in UTC

Reply:
I am afraid I don't understand what you mean by supporting this feature in IST. The test result is correct for UTC and any offset such as `xx:20`/`xx:40`/`xx:00` (`xx:20`/`xx:40` probably doesn't exist).

In case for IST or any offset such as `xx:30`, the correct answer would be 70.

![Explanation](https://github.com/calcom/cal.com/assets/86384652/5530d8e6-27ca-45e6-831c-cb2be3bb9378)

In UTC the correct result would be 5. In Iran the correct result is 4.

This is because the slots interval takes attendee tz in mind.

Reply:
The higher the interval, the more prominent the issue.
Currently we support these intervals `60` `30` `20` `15` `10`.
GCF of all these is `5` which will solve the issue but that would be a performance issue.

Reply:
`getSlots` fn takes `dateRanges` in the host's tz.

Reply:
`dayjs()` will return the dates in the local tz causing the test to fail in tz other than UTC.